tools/evm/Program.cs

```diff
@@ -36,6 +36,7 @@ static void ConfigureT8NCommand(ref RootCommand rootCmd)
             var traceOpt = new Option<bool>("--trace", description: "Configures the use of the JSON opcode tracer. This tracer emits traces to files as trace-<txIndex>-<txHash>.jsonl", getDefaultValue: () => false);
             var traceNoStackOpt = new Option<bool>("--trace.noStack", description: "Trace no stack", getDefaultValue: () => false);
             var traceReturnDataOpt = new Option<bool>("--trace.returnData", description: "Trace return data", getDefaultValue: () => false);
+            var isGnosisTests = new Option<bool>("--run.gnosis", description: "If set to true, will run under Gnosis spec", getDefaultValue: () => false);
```

Comment:
Just an idea for an improvement: instead of using a new `--run.gnosis` option maybe we can reuse  `--state.chainId` and choose `ISpecProvider` based on chain / [networkid](https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Chains/gnosis.json#L44)? With such a way we will not need to add a new flag every time we have to add a new network to t8n.

Reply:
Good idea, removed --run.gnosis flag